### PR TITLE
(#3385) OUTDATED NCIDS for Application Modules

### LINKED ISSUE #3385: Story: Update Application Pages for NCIDS Layout
## As a user, I would like to view Application Page content using the NCIDS, so that I have a better navigation experience.

This ticket is to update the application page templates which use both a full-page layout without left navigation and a layout with left navigation.

As part of this ticket, the Inner Pages background color should be updated to be #FFFFFF

> **ESTIMATE** 8

### Acceptance Criteria

```gherkin
Scenario: [title]
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] #3240 

### Tasks
- [ ] templates/content/application_page/page--cgov-application-page--no-left-nav.html.twig
  - https://www.cancer.gov/search/results?swKeyword=cancer
  - https://www.cancer.gov/about-cancer/treatment/clinical-trials/search
  - https://www.cancer.gov/research/resources
- [ ] templates/content/application_page/page--cgov-application-page--full.html.twig
  - https://www.cancer.gov/publications/dictionaries/cancer-terms
  - https://www.cancer.gov/publications/dictionaries/cancer-drug
  - https://www.cancer.gov/publications/dictionaries/genetics-dictionary

- [ ] update all app modules (new task?)
- [ ] templates/app_module/r4r/app-module--cgov-r4r-app.html.twig
- [ ] templates/layout/html--cgov-application-page.html.twig:
   - [ ] replace skipnav with `.usa-skipnav` 

#### Node templates
- [ ] templates/content/application_page/node--cgov-application-page--full.html.twig

### Technical Notes
- CTS should be moved to be a generic app module by the time we get here.

### LINKED ISSUE #3522: Update CTS LiveHelp pop-up for NCIDS Layouts
### Issue description
The Proactive Chat pop-up is currently missing styling on the inner and application pages on which it shows. The pop-up styles need to be updated to work in NCIDS.

This is related to #3385 and #3379 and maybe #3380 

> **ESTIMATE** TBD

### Steps to reproduce the issue

1.  
2. 
3. 

### What's the expected result?
- The LiveHelp pop-up displays with it's current CGov styling.

### What's the actual result?
- The LiveHelp pop-up is smushed into the bottom left corner, with no styling.

### Additional details / screenshot
livehelp

### Related Tickets
* Issue #9999
* Issue #9999

### PR DESCRIPTION
**Pull request by bryanpnih**
ODE: http://ncigovcdode479.prod.acquia-sites.com
Closes #3385
Closes #3522

Note: The R4R pages will still have the external link icon for non-external links when viewed on an ODE or locally due to the way the regex statement is formatted. The regex statement checks if the URL is a .gov, hence it will not work on our .com ODE.

### CODE REVIEW

**Comment by bennettcc** [comment]
@bryanpnih Here's the first round of feedback on application pages. Feedback on CTS and the listing pages will be added later.

**NCI Dictionary of Cancer Terms**
1)	This is probably not connected to the app work, but I clicked on this link from the dictionary: https://ncigovcdode479.prod.acquia-sites.com/syndication/widgets and the image is missing (to the right of the code box, see live site: https://www.cancer.gov/syndication/widgets)
2)	Individual page – sound icon is too close to text: https://ncigovcdode479.prod.acquia-sites.com/publications/dictionaries/cancer-terms/def/ibandronate (see live site: https://www.cancer.gov/publications/dictionaries/cancer-terms/def/ibandronate)

**NCI Dictionary of Genetics Terms**
All good - no issues

**NCI Drug Dictionary**
1)	Individual page – sound icon is too close to text: https://ncigovcdode479.prod.acquia-sites.com/publications/dictionaries/genetics-dictionary/def/acrochordon (see live site: https://www.cancer.gov/publications/dictionaries/genetics-dictionary/def/acrochordon)

**R4R** 
https://ncigovcdode479.prod.acquia-sites.com/research/resources
1)	Probably a nonissue: On the R4R search results page, the R4R Home link and each resource link has the exit disclaimer. Is this a product of being on the ODE so the URLs aren’t directing to a .gov?

**Sitewide search** 
https://ncigovcdode479.prod.acquia-sites.com/search/results/?swKeyword=cancer
1)	After clicking the sound icon to play the pronunciation – the blue active-state outline covers part of the word ‘cancer’
2)	There is a lot of space between the page option numbers as it expands to the left (to reproduce, go to page 4). This causes the Results count and searched term to wrap.
 3)	Show # Results arrow partially covers the number of pages at bottom of the search results page

**Comment by sarinapadilla** [comment]
**Dynamic/Manual Listing Pages**
1. The pager is spaced very wide and wonky. Once you get to page 2, the addition of the `< Previous` juts up to the "Trials 26-50 of x" and causes it to wrap. Going even further through the pager continues to add to the spacing/text wrapping issues. It looks less bad on tablet and mobile since the pager is moved to its own section, but the spacing causes it to wrap much earlier than it does on Prod.
2. The breadcrumb displays "Disease Dynamic Trial Listing Page" (or "Intervention Dynamic Trial Listing Page") because that is the level you are currently on and its name in the system. This is expected behavior, but logging that we are going to have to come up with a better name for that since it's going to show on *all* disease and intervention pages.
3. There is too much spacing between 1) trial names and descriptions, and 2) trial descriptions and locations, for each listed trial.
4. Because of the spacing above, there is no clear distinction between each of the listed trials. The space between the description and location of one trial is the same as the space between the location and the next listed trial.
5. The LiveHelp popup has basically no styling. Doesn't float on top of the content, but rather is tacked on at the bottom of the page; text doesn't wrap; there's essentially no styling. Clicking the Chat Now and the x both work.

**Comment by sarinapadilla** [comment]
**CTS**
Where to begin...

Basic Form:
1. The borders on the form fields are all kinds of broken. The top border isn't being displayed, and the sides don't extend above the form field name.
2. The spacing between the Cancer Type/Keyword form field name and text input are too close together. This also causes the text input to overlap onto the help link.
3. Autosuggest has a small space between the text input and the suggestions.
4. The clinical trials API banner floats at the bottom. This is expected, as it's what happens on Prod, but it looks...weird. Likely because we don't have a defined content area like on current CTS.

Advanced Form:
1. Same issue with form field borders as above.
2. In the Primary Cancer Type field, once a main type is selected, the inputs for Subtype, Stage, and Findings have double bordering and text box sizing issues.
3. In the Location form field, the toggle for the VA option on locations is being cut off.
4. In the Location form field, the spacing between the options is much too close. The selected option's radio button gets an outline that overlaps on the other radio buttons.
5. In the Location form field with Country/State/City option selected, the State input has double bordering and text box sizing issues.
6. In the Trial Type form field, the description of the field is overlapping onto the help button.
7. In the Trial Type form field, the healthy volunteers toggle is getting cut off.
8. In the Trial Type form field, the spacing between the options is too close. When an option's checkbox is actively selected, it gets an outline that almost overlaps the other checkboxes.
9. In the Drug/Treatment form field, the inputs for Drug/Drug Family and Other Treatments have double bordering and text box sizing issues.
10. In the Trial Phase form field, the spacing between the options is too close. When an option's checkbox is actively selected, it gets an outline that almost overlaps the other checkboxes.
11. In the Trial ID form field, the description of the field is overlapping onto the help button.
12. In the Trial Investigators form field, the description of the field is overlapping onto the help button.
13. In the Lead Organization form field, the description of the field is overlapping onto the help button.

**Comment by kate-mashkina** [comment]
Two things that still persist 
CTS Advanced Form:
 toggle is cut off from the bottom at desktop and tablet , at 320 px it is also being cut off from the right side

Screen Shot 2022-09-06 at 1 01 03 PM
Screen Shot 2022-09-06 at 12 59 23 PM
Glossary:
Pronunciation button is too close to the text
Screen Shot 2022-09-06 at 12 34 56 PM

**Comment by bennettcc** [comment]
@bryanpizzillo One more issue on sitewide search:

When clicking the pronunciation button of the searched term, the active state blue outline (around the megaphone) covers a part of that term

image

**Comment by blilianyu** [comment]
**General**
- on Safari, search icon is not showing in mobile menu 
Screen Shot 2022-09-07 at 3 45 48 PM


**CTS**
- on Chrome, Firefox & Safari, clinical trial modal seems to be appearing at the bottom below pagination
Screen Shot 2022-09-07 at 3 37 21 PM
Screen Shot 2022-09-07 at 3 38 40 PM


**Dynamic Listing Pages & Manual Listing Pages**
- on Chrome, Firefox & Safari, pagination spacing is larger than CTS & R4R
Screen Shot 2022-09-07 at 3 44 36 PM

**Glossaries**
- Chrome & Firefox, Speaker icon is extremely large and the focus cuts it off 
Screen Shot 2022-09-07 at 3 49 05 PM
- Safari, speaker icon is cut off
Screen Shot 2022-09-07 at 3 50 27 PM
- Chrome, Firefox & Safari, speaker icon is running into pronunciation
Screen Shot 2022-09-07 at 3 50 59 PM

**Sitewide Search**
- Chrome, Firefox & Safari, pagination spacing is larger than CTS & R4R
Screen Shot 2022-09-07 at 3 57 14 PM

**R4R**
- Chrome, Firefox & Safari, there is an external link icon after every item when there shouldn't be and after R4R home link
Screen Shot 2022-09-07 at 3 59 44 PM
Screen Shot 2022-09-07 at 4 02 21 PM

**Comment by bryanpizzillo** [comment]
>This is probably not connected to the app work, but I clicked on this link from the dictionary: https://ncigovcdode479.prod.acquia-sites.com/syndication/widgets and the image is missing (to the right of the code box, see live site: https://www.cancer.gov/syndication/widgets)
So 2 things, 
1. this only works on ACSF, and maybe only live.
2. This is not an image but the actual working widget.

Gonna call this something to test in ACSF.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/r4r/exitDisclaimerHandler.js [hunk 1]
```diff
@@ -20,10 +20,11 @@ export const exitDisclaimerInjector = () => {
   const externalLinks = Array.from(allLinks).filter(link => {
     if (link.href !== "") {
       return (
-        /^https?:\/\/([a-zA-Z0-9-]+\.)+/i.test(link.href) &&
+        /^https?:\/\/([a-zA-Z0-9-]+)+/i.test(link.href) &&
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/r4r/exitDisclaimerHandler.js:23]: undo the changes to this file

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/application_page/page--cgov-application-page--left-nav.html.twig [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{# CHANGE THIS TO ncids-global when updateing for ncids #}
+{{ attach_library('ncids_trans/ncids-global') }}
+
+{{ page.ncids_banner }}
+<!-- NOTIFICATIONS HERE -->
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/application_page/page--cgov-application-page--left-nav.html.twig:5]: add site alerts

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/application_page/page--cgov-application-page--no-left-nav.html.twig [hunk 3]
```diff
@@ -1,58 +1,33 @@
 {# CHANGE THIS TO ncids-global when updateing for ncids #}
-{{ attach_library('ncids_trans/legacy-global') }}
+{{ attach_library('ncids_trans/ncids-global') }}
 
-{{ page.notification_banner }}
+{{ page.ncids_banner }}
+<!-- NOTIFICATIONS HERE -->
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/content/application_page/page--cgov-application-page--no-left-nav.html.twig:5]: add site alerts

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Comment by bennettcc** [comment]
Internal testing round 3, beginning 9/19

**Comment by dev-rana-publicis** [comment]
Note: The R4R pages will still have the external link icon for non-external links when viewed on an ODE or locally due to the way the regex statement is formatted. The regex statement checks if the URL is a .gov, hence it will not work on our .com ODE.

**Comment by laurelthrash** [comment]
Issue related to CTS Proactive Chat Pop-up: Identified during testing on 09/19. The body text in the proactive chat pop-up maintains the former body style (screen shot). Cc: @olitharp-nci
image

**Pull request closed without merging**